app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,239 @@
+package com.brandon.forzenbook.viewmodels
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.core.text.isDigitsOnly
+import androidx.core.util.PatternsCompat
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.brandon.forzenbook.usecase.CreateUserUseCase
+import com.brandon.forzenbook.usecase.GetLoginCodeUseCase
+import com.brandon.forzenbook.usecase.LoginUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.sql.Date
+import java.text.SimpleDateFormat
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase,
+    private val getLoginCodeUseCase: GetLoginCodeUseCase,
+    private val createUserUseCase: CreateUserUseCase,
+) : ViewModel() {
+
+    private var sqlDate: Date? = null
+
+    private val _uiState: MutableState<LoginUiState> = mutableStateOf(LoginUiState.Idle())
+    val uiState: MutableState<LoginUiState>
+        get() = _uiState
+
+    private val _createAccountState: MutableState<CreateAccountUiState> =
+        mutableStateOf(CreateAccountUiState.Idle())
+    val createAccountState: MutableState<CreateAccountUiState>
+        get() = _createAccountState
+
+    sealed interface LoginUiState {
+
+        data class Idle(
+            val email: String = "",
+            val isEmailError: Boolean = false,
+            val isGenericError: Boolean = false,
+            val isCodeSent: Boolean = false,
+        ) : LoginUiState
+
+        data class Loading(
+            val email: String = "",
+            val isCodeSent: Boolean = false,
+        ) : LoginUiState
+
+        // TODO remove this once we have another screen FA-83
+        object LoginWithCode : LoginUiState
+
+    }
+
+    sealed interface CreateAccountUiState {
+
+        data class Idle(
+            val email: String = "",
+            val firstName: String = "",
+            val lastName: String = "",
+            val location: String = "",
+            val dateOfBirth: String = "",
+            val isEmailError: Boolean = false,
+            val isLocationError: Boolean = false,
+            val isDateError: Boolean = false,
+            val isDuplicateUser: Boolean = false,
+        ) : CreateAccountUiState
+
+        data class Loading(
+            val email: String = "",
+            val firstName: String = "",
+            val lastName: String = "",
+            val location: String = "",
+            val dateOfBirth: String = "",
+        ) : CreateAccountUiState
+
+        object Loaded : CreateAccountUiState
+    }
+
+    fun getCode(email: String) {
```

Comment:
This is a public method not defined by a UI action

Reply:
renamed